control/prometheus.py

```diff
@@ -0,0 +1,196 @@
+            "BDEV Capacity", 
+            labels=["bdev_name"])
+
+        for bdev in bdev_info:
+            bdev_name = bdev.get('name')
+            try:
+                rbd_info = bdev["driver_specific"]["rbd"]
+            except KeyError:
+                self.logger.debug(f"no rbd information present for bdev {bdev.get('name')}, skipping")
+                continue
+
+            rbd_pool = rbd_info.get('pool_name')
+            rbd_image = rbd_info.get('rbd_name')
+            if self.bdev_pools:
+                if rbd_pool not in self.bdev_pools:
+                    continue
+
+            bdev_lookup[bdev_name] = RBD(rbd_pool, rbd_image)
+            bdev_metadata.add_metric([bdev_name, rbd_pool, rbd_image], 1)
+            bdev_size = bdev.get("block_size") * bdev.get("num_blocks")
+            bdev_capacity.add_metric([bdev.get("name")], bdev_size)
+        
+        yield bdev_capacity
+        yield bdev_metadata
+     
+        bdev_io_stats = rpc.bdev.bdev_get_iostat(self.spdk_rpc_client)
+
+        tick_rate = bdev_io_stats.get("tick_rate")
+
+        bdev_read_ops = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_reads_completed_total", 
+            "Total number of read operations completed", 
+            labels=["bdev_name"])
+        bdev_write_ops = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_writes_completed_total", 
+            "Total number of write operations completed", 
+            labels=["bdev_name"])
+        bdev_read_bytes = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_read_bytes_total", 
+            "Total number of bytes read successfully", 
+            labels=["bdev_name"])
+        bdev_write_bytes = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_written_bytes_total", 
+            "Total number of bytes written successfully", 
+            labels=["bdev_name"])
+        bdev_read_seconds = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_read_seconds_total", 
+            "Total time spent servicing READ I/O", 
+            labels=["bdev_name"])
+        bdev_write_seconds = CounterMetricFamily(
+            f"{self.metric_prefix}_bdev_write_seconds_total", 
+            "Total time spent servicing WRITE I/O", 
+            labels=["bdev_name"])
+
+        for bdev in bdev_io_stats.get("bdevs", []):
+            bdev_name = bdev.get('name')
+            if bdev_name not in bdev_lookup:
+                self.logger.debug(f"i/o stats for bdev {bdev_name} skipped. Either not an rbd bdev, or excluded by 'prometheus_bdev_pools'")
+                continue
+            
+            bdev_read_ops.add_metric([bdev_name], bdev.get("num_read_ops", 0))
+            bdev_write_ops.add_metric([bdev_name], bdev.get("num_write_ops", 0))
+            bdev_read_bytes.add_metric([bdev_name], bdev.get("bytes_read", 0))
+            bdev_write_bytes.add_metric([bdev_name], bdev.get("bytes_written", 0))
```

Comment:
I'm curious why 0 is passed here, but isn't passed for "latency" below.  Can "ops" and "bytes" counters not be there, but "latency" counters are guaranteed to be present?